rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/LocalInfrastructure.java

```diff
@@ -119,19 +120,23 @@ public void acquireAllNodes() {
 
     @Override
     public void acquireNode() {
-        startNodes(1);
+        if (maxNodes - getNumberOfAcquiredNodesWithLock() > 0) {
+            startNodes(1);
+        }
     }
 
     @Override
     public void acquireNodes(int n, Map<String, ?> nodeConfiguration) {
-        if (n > 0) {
+        if (n > 0 && (maxNodes - getNumberOfAcquiredNodesWithLock() - n) >= 0) {
```

Comment:
this is barely readable, what is n?

Reply:
OK changed in 38b2ed2f5df5328736c051dcc52116bc112cf27d, BTW this is the actual signature of the overrided NodeSource method.